### PR TITLE
Log cluster.initial_master_nodes at startup

### PR DESCRIPTION
**Pull request by DaveCTurner**
We continue to see occasional cases of clusters with inappropriate
bootstrap config. It can be quite hard to diagnose these cases because
we don't report the bootstrap config anywhere in the logs. This commit
adds that missing logging to simplify the diagnostic process.

### CODE REVIEW

**Review by arteam — APPROVED** [review]
LGTM! Love the extensive tests of logging

File: server/src/test/java/org/elasticsearch/cluster/coordination/ClusterBootstrapServiceTests.java [hunk 1]
```diff
@@ -648,9 +654,81 @@ public void testFailBootstrapNonMasterEligibleNodeWithSingleNodeDiscovery() {
...
+                )
+            );
+
+            new ClusterBootstrapService(
+                Settings.builder().putList(INITIAL_MASTER_NODES_SETTING.getKey(), "node1", "node2").build(),
+                transportService,
+                Collections::emptyList,
+                () -> false,
+                vc -> { throw new AssertionError("should not be called"); }
+            ).logBootstrapState(Metadata.builder().clusterUUIDCommitted(false).build());
```
Comment (henningandersen) [inline server/src/test/java/org/elasticsearch/cluster/coordination/ClusterBootstrapServiceTests.java:685]: Can we randomly add a `clusterUUID` too?

**Review by henningandersen — APPROVED** [review]
LGTM.

File: server/src/test/java/org/elasticsearch/cluster/coordination/ClusterBootstrapServiceTests.java (hunk 1 above)
Comment (DaveCTurner) [reply server/src/test/java/org/elasticsearch/cluster/coordination/ClusterBootstrapServiceTests.java]: Indeed we can, see [9e17662](https://github.com/elastic/elasticsearch/pull/86101/commits/9e1766286493e4979f4d33f3396da1bd9ab838b4).

**Pull request merged**